tianshou/data/utils.py

```diff
@@ -37,7 +37,7 @@ def to_torch(x: Union[torch.Tensor, dict, Batch, np.ndarray],
     elif isinstance(x, (np.number, np.bool_, Number)):
         x = to_torch(np.asanyarray(x), dtype, device)
     elif isinstance(x, list) and len(x) > 0 and \
-            isinstance(x[0], (np.number, np.bool_, Number)):
+            all(isinstance(e, (np.number, np.bool_, Number)) for e in x):
```

Comment:
Same.

Reply:
change to `isinstance(x[0], (np.number, np.bool_, Number)) and all(isinstance(e, (np.number, np.bool_, Number)) for e in x)` because `all` will calculate all items in the list even if the first one cannot pass the check

Reply:
According to [here](https://stackoverflow.com/questions/2580136/does-python-support-short-circuiting), python ``all`` supports short circuiting.